File: 01_Language/01_Functions/php/natcasesort.php
<?php

$array1 = $array2 = array('IMG0.png', 'img12.png', 'img10.png', 'img2.png', null, 'img1.png', 'IMG3.png');

sort($array1);
echo "Standard sorting\n";
print_r($array1);

natcasesort($array2);
echo "\nNatural order sorting (case-insensitive)\n";
print_r($array2);

$array3 = ['啊', '波', '吃', '爱', '笔', '虫'];
natcasesort($array3);
print_r($array3);

File: 01_Language/01_Functions/python/natcasesort.py
# coding: utf-8

def natcasesort(array):
    array.sort(key=lambda x: '' if x is None else str(x).lower())


if __name__ == '__main__':
    array1 = array2 = ['IMG0.png', 'img12.png', 'img10.png', 'img2.png', None, 'img1.png', 'IMG3.png']
    natcasesort(array1)
    print(array1)
